openzim/fcctozim/build.py

```diff
@@ -1,66 +1,95 @@
 import json
+import os
+from collections import OrderedDict
+
+from fcctozim import VERSION, FCCLangMap, logger
+from zimscraperlib.zim import Creator
 
-from zimscraperlib.zim import Archive, Creator, StaticItem, URLItem
-from frontmatter import Frontmatter
-from fcctozim import logger
+logo_path = os.path.join(os.path.dirname(__file__), "..", "fcc_48.png")
 
-logo_path = os.path.join(os.path.dirname(__file__), '..', 'fcc_48.png')
 
 def build_curriculum_redirects(clientdir, language):
-    index_json_path = os.path.join(clientdir, 'src/assets/fcc/index.json')
+    fcc_lang = FCCLangMap[language]
+    index_json_path = os.path.join(clientdir, "fcc/index.json")
     with open(index_json_path) as course_index_str:
-        course_list = json.load(course_index_str)[language]
+        course_list = json.load(course_index_str)[fcc_lang]
 
     redirects = []
     for course in course_list:
-        meta_json_path = os.path.join(clientdir, 'src/assets/fcc/curriculum/', language, course, '_meta.json')
+        meta_json_path = os.path.join(
+            clientdir, "fcc/curriculum/", fcc_lang, course, "_meta.json"
+        )
         with open(meta_json_path) as meta_json_str:
-            challenges = json.load(meta_json_str)['challenges']
+            challenges = json.load(meta_json_str)["challenges"]
         for challenge in challenges:
-            redirects.append([ f'index.html#{language}/{course}/{challenge["slug"]}', challenge['title'] ])
-            
-    return redirects
+            title = challenge["title"]
+            redirects.append((f'{fcc_lang}/{course}/{challenge["slug"]}', title))
 
+    return OrderedDict(redirects).items()
 
-def build_zimfile(clientdir, outpath, language):
-    source_dir = os.path.join(clientdir, 'dist')
-    rootPath = os.path.join(source_dir, 'index.html')
+
+def build(arguments):
+    clientdir = arguments.clientdir
+    outpath = arguments.outzim
+    language = arguments.language
+    name = arguments.name
+    title = arguments.title
+    description = arguments.description
+
+    logger.info(
+        f"Building {clientdir} for {language} => {outpath} - Version: {VERSION}"
+    )
+
+    source_dir = os.path.join(clientdir)
+    rootPath = os.path.join(source_dir, "index.html")
     fileList = []
-    for (root, dirs, files) in os.walk(source_dir):
-        for name in files:
-            fileList.append(os.path.join(root, name))
 
-    fileList = [file for file in fileList if os.path.splitext(file)[1] != '.map']
+    # Walk the ree and get a list of files we care about
+    for root, dirs, files in os.walk(source_dir):
+        for name in files:
+            file = os.path.join(root, name)
+            if os.path.splitext(file)[1] == ".map":
+                continue
+            fileList.append(file)
 
     main_path = os.path.relpath(rootPath, source_dir)
+
+    # Make sure the outpath directory exists
+    outpathExists = os.path.exists(os.path.dirname(outpath))
+    if not outpathExists:
```

Comment:
Refatored with https://github.com/openzim/freecodecamp/commit/309eec193d38b41a2deb326079ec06a6105fa308#diff-b914a73f366994735aa1a6acad1ce82ec8faca47a68294c836f987674ffa0f90R59

`pathlib.Path(outpath).mkdir(parents=True, exist_ok=True)` to avoid the if check